### PR TITLE
feat(io): add a xcp_fd_preadv function

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

File: include/xcp-ng/generic/io.h [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #define _XCP_NG_GENERIC_IO_H_
 
 #include <poll.h>
+#include <sys/uio.h>
```
Comment (benjamreis) [inline include/xcp-ng/generic/io.h]: Can't you do a forward declaration and include in the .c ?

Comment (Wescoeur) [reply include/xcp-ng/generic/io.h]: Yes I can.

File: src/io.c [hunk 2]
```diff
@@ -156,9 +157,22 @@ XcpError xcp_fd_pread (int fd, void *buf, size_t count, off_t offset) {
...
+
+  return XCP_ERR_ERRNO;
+}
+
 // -----------------------------------------------------------------------------
 
-XcpError xcp_poll (struct pollfd *fds, nfds_t nfds, int timeout) {
+XcpError xcp_poll (struct pollfd *fds, uint nfds, int timeout) {
   do {
     const int ret = poll(fds, nfds, timeout);
```
Comment (benjamreis) [inline src/io.c:177]: Does the call build without casting nfds to `nfds_t` ?
IF it does we need to add an implicit cast warning i think.

Comment (Wescoeur) [reply src/io.c:177]: No compilation problem. I think it's useless to add a cast if there is no warning about this.

**Comment by benjamreis** [comment]
Ok

**Pull request merged**